### PR TITLE
Task/upgrade shares

### PR DESCRIPTION
**Pull request by triggerNZ**
Series of commits to upgrade shares to V2 done by @vijayshiyani and @triggerNZ.

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -9,10 +9,10 @@
...
   "bugs": "https://github.com/Meeco/cli/issues",
   "dependencies": {
-    "@meeco/cryppo": "^0.7.0",
-    "@meeco/keystore-api-sdk": "^0.14.1-beta.3",
-    "@meeco/sdk": "^0.0.12-alpha.2",
-    "@meeco/vault-api-sdk": "^0.19.1-beta.4",
+    "@meeco/cryppo": "^0.7.1",
+    "@meeco/keystore-api-sdk": "0.14.1-beta.1",
+    "@meeco/sdk": "0.0.12-alpha.2",
+    "@meeco/vault-api-sdk": "0.19.1-beta.3",
```
Comment (zbarbuto) [inline packages/cli/package.json:15]: Any particular reason these got downgraded?

Comment (vijayshiyani) [reply packages/cli/package.json:15]: not really, I think it was reverted during the merge. I will fix it.

> Any particular reason these got downgraded?

File: packages/sdk/package.json [hunk 2]
```diff
@@ -9,9 +9,9 @@
   "module": "dist/index.esm.js",
   "types": "dist/src/index.d.ts",
   "dependencies": {
-    "@meeco/cryppo": "^0.7.0",
-    "@meeco/keystore-api-sdk": "^0.14.1-beta.3",
-    "@meeco/vault-api-sdk": "^0.19.1-beta.4",
+    "@meeco/cryppo": "^0.7.1",
+    "@meeco/keystore-api-sdk": "0.14.1-beta.1",
+    "@meeco/vault-api-sdk": "0.19.1-beta.3",
```
Comment (vijayshiyani) [inline packages/sdk/package.json:14]: @zbarbuto same here, I will update this as well

File: packages/cli/src/commands/shares/update.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    const authConfig = await this.readConfigFromFile(AuthConfig, auth);
+
+    if (!authConfig) {
+      this.error('Valid auth config file must be supplied');
+    }
+
+    const service = new ShareService(environment, this.updateStatus);
+    try {
+      const reslut = await service.updateSharedItem(authConfig, itemId);
```
Comment (zbarbuto) [inline packages/cli/src/commands/shares/update.ts:37]: > `reslut`
😂

File: packages/cli/src/commands/shares/get-incoming.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    const authConfig = await this.readConfigFromFile(AuthConfig, auth);
+
+    if (!authConfig) {
+      this.error('Valid auth config file must be supplied');
+    }
+
+    const service = new ShareService(environment, this.updateStatus);
+    try {
+      const reslut = await service.getSharedItemIncoming(authConfig, shareId);
```
Comment (zbarbuto) [inline packages/cli/src/commands/shares/get-incoming.ts:38]: typo

File: packages/sdk/src/services/item-service.ts [hunk 5]
```diff
@@ -141,11 +163,7 @@ export class ItemService {
...
 
     this.logger.log('Fetching item');
-    const itemFetchResult = await this.get(
-      itemId,
-      auth.vault_access_token,
-      auth.data_encryption_key
-    ).catch(err => {
+    const itemFetchResult = await this.get(itemId, auth).catch(err => {
       throw new MeecoServiceError(
         `Unable to find item '${itemId}' - please check that the item exists for the current user.`
```
Comment (zbarbuto) [inline packages/sdk/src/services/item-service.ts:168]: This isn't part of the changes but I'm pointing it out now while I see it to fix later - we probably should actually be checking the status code here before blindly throwing "unable to find item".
It should only throw this error if it's literally a 404.

File: packages/cli/src/commands/shares/update.ts (hunk 3 above)
Comment (vijayshiyani) [reply packages/cli/src/commands/shares/update.ts]: fixed

File: packages/cli/src/commands/shares/get-incoming.ts (hunk 4 above)
Comment (vijayshiyani) [reply packages/cli/src/commands/shares/get-incoming.ts]: fixed

File: packages/sdk/src/services/item-service.ts (hunk 5 above)
Comment (vijayshiyani) [reply packages/sdk/src/services/item-service.ts]: fixed

**Pull request merged**